server/rootless.go

```diff
@@ -45,3 +83,27 @@ func makeOCIConfigurationRootless(g *generate.Generator) {
 
 	g.SetLinuxCgroupsPath("")
 }
+
+// getAvailableV2Controllers returns the entries in /sys/fs/cgroup/<SELF>/cgroup.controllers
+func getAvailableV2Controllers() map[string]struct{} {
+	procSelfCgroup, err := cgroups.ParseCgroupFile("/proc/self/cgroup")
+	if err != nil {
+		logrus.WithError(err).Error("failed to parse /proc/self/cgroup")
+		return nil
+	}
+	v2Group := procSelfCgroup[""]
+	if v2Group == "" {
+		return nil
+	}
+	controllersPath := filepath.Join("/sys/fs/cgroup", v2Group, "cgroup.controllers")
+	controllersBytes, err := ioutil.ReadFile(controllersPath)
+	if err != nil {
+		logrus.WithError(err).Errorf("failed to read %s", controllersPath)
```

Comment:
nit: `ioutil.ReadFile` error message already contains file name, so no need to wrap an error.
```suggestion
		logrus.Error(err)
```

Reply:
updated